test/testdata/input/splicelist_in_nested_block_list.test

```diff
@@ -0,0 +1,7 @@
+SLICE LIST IN NESTED BLOCK LIST
+---
+key:
+- true
+- - false
```

Comment:
Could we make an example with even more nesting :rofl: 

What about:
```yaml
key:
 - foo:
   - true
   - - bar:
       - true
       - - false
```
or
```yaml
key:
 - - foo
```
or
```yaml
key:
 - true
 - - true
 - - - false
```
or
```yaml
key:
 - - - false
```
or
```yaml
key:
 - - - false
 - - - false
 - - - false
```

uh, there are so many degenerate things one can do :rofl:

Reply:
I have some of these tests locally. Lemme push. I'll create more golden tests with these suggestions 😅

Reply:
Yeah, but all means feel free to spam with test files :rocket: 

When they are written this way, it's pretty easy to update the tests if someday the API changes..